kernel/src/transaction.rs

```diff
@@ -175,6 +174,13 @@ impl Transaction {
         self
     }
 
+    /// Add commit info to this transaction and append the argument as its engine_commit_info.
+    /// The engine is required to provide commit info before committing the transcation.
```

Comment:
need to update the comment. I think neither is true anymore (should say we add _engine_ info within commit info and note that now this isn't required)